statsmodels/regression/linear_model.py

```diff
@@ -1902,16 +1902,19 @@ def eigenvals(self):
         if self._wexog_singular_values is not None:
             eigvals = self._wexog_singular_values ** 2
         else:
-            eigvals = np.linalg.linalg.eigvalsh(np.dot(self.model.wexog.T,
-                                                       self.model.wexog))
+            wx = self.model.wexog
+            eigvals = np.linalg.linalg.eigvalsh(wx.T @ wx)
         return np.sort(eigvals)[::-1]
 
     @cache_readonly
     def condition_number(self):
         """
         Return condition number of exogenous matrix.
 
-        Calculated as ratio of largest to smallest eigenvalue.
+        Calculated as ratio of largest to smallest eigenvalue of the exogenous
```

Comment:
instead of eigenvalue it should say singular value.
AFAIK, eigenvalues are only defined for square matrices